caas/kubernetes/clientconfig/k8s.go

```diff
@@ -162,3 +161,60 @@ func getKubeConfigPath() string {
 	}
 	return envPath
 }
+
+// getKubeConfig reads kubeconfig file either stdin or config file, then deserializes the contents into Config object.
+func getKubeConfig() (*clientcmdapi.Config, error) {
+	var filename string
+	var raw []byte
+
+	info, err := os.Stdin.Stat()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if info.Mode()&os.ModeNamedPipe != 0 || info.Size() > 0 {
+		// Try to read from Stdin.
+		raw, err = ioutil.ReadAll(os.Stdin)
+	} else {
+		// Try to read from kubeconfig file because no Stdin.
+		filename = getKubeConfigPath()
+		raw, err = ioutil.ReadFile(filename)
+		if err != nil {
+			err = errors.Annotatef(err, "failed to read kubernetes config from '%s'", filename)
+		}
+	}
+	if err != nil {
```

Comment:
Tghis err will only be populated in the if-stmt... Why not move it closer to the call? Say, right after ```ReadAll```

Reply:
i was thinking here just need return once rather than twice (one in if and one in else)